Gems/Profiler/Code/Source/ImGuiTreemap.cpp

```diff
@@ -0,0 +1,645 @@
+        m_lastExtent[0] = 0;
+        m_selectedNode = nullptr;
+        m_currentMask = 0xffffffff;
+    }
+
+    void ImGuiTreemapImpl::Render(int x, int y, int w, int h)
+    {
+        if (m_root.m_children.empty())
+        {
+            ImGui::Begin(m_name.IsEmpty() ? "Unnamed treemap" : m_name.GetCStr());
+            ImGui::End();
+            return;
+        }
+
+        ImVec2 offset{ static_cast<float>(x), static_cast<float>(y) };
+        ImGui::SetNextWindowPos(offset, ImGuiCond_Once);
+        ImVec2 extent{ static_cast<float>(w), static_cast<float>(h) };
+        ImGui::SetNextWindowSize(extent, ImGuiCond_Once);
+
+        if (ImGui::Begin(m_name.IsEmpty() ? "Unnamed treemap" : m_name.GetCStr()))
+        {
+            ImGui::Text("Total weight: %f %s", m_root.m_weight, m_unitLabel);
+            if (!m_groups.empty())
+            {
+                ImGui::Text("Highlight Group");
+            }
+            for (auto& group : m_groups)
+            {
+                ImGui::SameLine();
+                ImGui::Checkbox(group.first.GetCStr(), &group.second.m_active);
+            }
+
+            if (!m_masks.empty())
+            {
+                if (ImGui::RadioButton("Display All", m_currentMask == 0xffffffff))
+                {
+                    m_currentMask = 0xffffffff;
+                    m_dirty = true;
+                    m_lastExtent[0] = 0;
+                    m_selectedNode = nullptr;
+                }
+
+                for (auto& mask : m_masks)
+                {
+                    ImGui::SameLine();
+                    if (ImGui::RadioButton(mask.first, m_currentMask == mask.second))
+                    {
+                        m_currentMask = mask.second;
+                        m_dirty = true;
+                        m_lastExtent[0] = 0;
+                        m_selectedNode = nullptr;
+                    }
+                }
+            }
+
+            if (m_selectedNode)
+            {
+                ImGui::Text(
+                    "Selected node: %s (%f %s)", m_selectedNode->m_name.IsEmpty() ? "[unnamed]" : m_selectedNode->m_name.GetCStr(),
+                    m_selectedNode->m_weight, m_unitLabel);
+            }
+            else
+            {
+                ImGui::Text("No node selected");
+            }
+            ImGui::Separator();
+
+            ImDrawList* drawList = ImGui::GetWindowDrawList();
+            const auto [cx, cy] = ImGui::GetCursorPos();
+            const auto [wx, wy] = ImGui::GetWindowPos();
+            const auto [ww, wh] = ImGui::GetWindowSize();
+            bool focused = ImGui::IsWindowFocused();
+
+            // Add 20 pixel gutter to bottom
+            WeighAndComputeLayout(
+                static_cast<int>(cx + wx), static_cast<int>(cy + wy), static_cast<int>(ww - cx), static_cast<int>(wh - cy) - 20);
+
+            const auto [mx, my] = ImGui::GetMousePos();
+
+            AZStd::vector<TreemapNode*> tooltipNodes;

```

Comment:
Yea, this could allocate up to N pointers (where N is the maximum depth of the tree) when a mouse hovers over that element, but I'm being sloppy here and will move it. This is the only place I'm seeing where a per-frame allocation occurs.